src/DeviceInput/InputDevices/deviceSourceManager.ts

```diff
@@ -43,18 +44,63 @@ export class DeviceSourceManager {
      */
     public getDevices: () => ReadonlyArray<DeviceSource<DeviceType>>;
 
+    /**
+     * Enable handling of input events
+     * @hidden
+     */
+    public enableEvents: () => void;
+
+    /**
+     * Disable handling of input events
+     * @hidden
+     */
+    public disableEvents: () => void;
+
+    /**
+     * Dispose of DeviceSourceManager
+     */
+    public dispose: () => void;
+
     /**
      * Default constructor
      * @param engine Used to get canvas (if applicable)
      */
     constructor(engine: Engine) {
         this._deviceSourceManager = InternalDeviceSourceManager._Create(engine);
 
-        this.onDeviceConnectedObservable = this._deviceSourceManager.onDeviceConnectedObservable;
-        this.onInputChangedObservable = this._deviceSourceManager.onInputChangedObservable;
-        this.onDeviceDisconnectedObservable = this._deviceSourceManager.onDeviceDisconnectedObservable;
+        // Observables
+        this.onDeviceConnectedObservable = new Observable<DeviceSource<DeviceType>>((observer) => {
+            this.getDevices().forEach((device) => {
+                if (device) {
+                    this.onDeviceConnectedObservable.notifyObserver(observer, device);
+                }
+            });
+        });
+        this.onInputChangedObservable = new Observable<IDeviceEvent>();
+        this.onDeviceDisconnectedObservable = new Observable<DeviceSource<DeviceType>>();
+
+        this._deviceSourceManager.registerManager(this);
+
         this.getDeviceSource = this._deviceSourceManager.getDeviceSource;
         this.getDeviceSources = this._deviceSourceManager.getDeviceSources;
         this.getDevices = this._deviceSourceManager.getDevices;
+        this.enableEvents = this._deviceSourceManager.enableEvents;
+        this.disableEvents = this._deviceSourceManager.disableEvents;
+
+        this.dispose = (): void => {
+            // Null out observable refs
+            this._deviceSourceManager?.unregisterManager(this);
+            this.onDeviceConnectedObservable.clear();
+            this.onInputChangedObservable.clear();
+            this.onDeviceDisconnectedObservable.clear();
+            // Null out function refs
+            this.getDeviceSource = <T extends DeviceType>(deviceType: T, deviceSlot?: number) => { return null; };
```

Comment:
```suggestion
            this.getDeviceSource = () => null;
```

Reply:
Similarly for the following lines.

Reply:
simplified signature as per we discussed offline